src/telliot/submitter/base.py

```diff
@@ -4,20 +4,91 @@
 in the `reporter_plugins` module. Submitters submit off-chain data on-chain
 to the Tellor oracle.
 """
+import json
 from abc import ABC
-from abc import abstractmethod
 from typing import Any
+from typing import Mapping
+from typing import Sequence
+
+import requests
+from telliot.reporter.config import ReporterConfig
+from telliot.utils.abi import tellor_playground_abi
+from telliot.utils.rpc_endpoint import RPCEndpoint
 
 
 class Submitter(ABC):
-    """Base class for a Submitter.
+    """Submits BTC on testnet.
+
+    Submits BTC price data in USD to the TellorX playground
+    on the Rinkeby test network."""
+
+    def __init__(
+        self, config: ReporterConfig, abi: Sequence[Mapping[str, Any]]
+    ) -> None:
+        """Reads user private key and node endpoint from `.env` file to
+        set up `Web3` client for interacting with the TellorX playground
+        smart contract."""
+        self.config = config
+
+        self.endpt = RPCEndpoint(
+            network=self.config.network,
+            provider=self.config.provider,
+            url=self.config.node_url,
+        )
+
+        self.endpt.connect()
+
+        self.acc = self.endpt.web3.eth.account.from_key(self.config.private_key)
+
+        self.contract = self.endpt.web3.eth.contract(
+            self.config.contract_address, abi=tellor_playground_abi
+        )
+
+    def build_tx(self, value: bytes, request_id: str, gas_price: str) -> Any:

```

Comment:
Hopefully things like this get replaced with Tally's "standardized" contract writes, so that we have uniform exeption handling, etc.